src/Features/CSharp/Portable/CodeRefactorings/CSharpRefactoringHelpersService.cs

```diff
@@ -58,5 +59,34 @@ protected override IEnumerable<SyntaxNode> ExtractNodesSimple(SyntaxNode? node,
                 }
             }
         }
+
+        protected override bool TryGetVariableDeclaratorInSingleFieldDeclaration(SyntaxNode node, [NotNullWhen(true)] out SyntaxNode? singleVariableDeclarator)
+        {
+            // 1. If node is a part of VariableDeclaration, check the grand parent of this node.
+            // e.g.
+            // class Bar
+            // {
+            //      public i$$nt i;
+            // }
+            if (node.Parent is VariableDeclarationSyntax && node.Parent.Parent is BaseFieldDeclarationSyntax { Declaration: { Variables: { Count: 1 } } } baseFieldDeclarationNode)
```

Comment:
can't the .Parent and .Parent.Parent parts be pulled into a single pattern check?